src/Microsoft.DotNet.Build.CloudTestTasks/CopyBlobsToLatest.cs

```diff
@@ -0,0 +1,242 @@
+            {
+                return false;
+            }
+
+            string sourceDir = $"{Product}/{ProductVersion}/";
+            string channelDir = $"{Product}/{Channel}/";
+            string semaphoreBlob = $"{channelDir}publishSemaphore";
+
+            CreateBlobIfNotExists(semaphoreBlob);
+
+            AzureBlobLease blobLease = new AzureBlobLease(
+                AccountName,
+                AccountKey,
+                ConnectionString,
+                ContainerName,
+                semaphoreBlob,
+                Log);
+
+            Log.LogMessage($"Acquiring lease on semaphore blob '{semaphoreBlob}'");
+            blobLease.Acquire();
+
+            try
+            {
+                if (EnableVersionHint)
+                {
+                    string targetVersionFile = $"{channelDir}{ProductVersion}";
+
+                    // Check if this build is already finalized.
+                    if (IsLatestSpecifiedVersion(targetVersionFile))
+                    {
+                        Log.LogMessage(
+                            MessageImportance.High,
+                            $"Version '{ProductVersion}' is already published. " +
+                                $"Skipping finalization. Hint file: '{targetVersionFile}'");
+
+                        return true;
+                    }
+
+                    // Delete old version files.
+                    GetBlobList(channelDir)
+                        .Select(s => s.Replace($"/{ContainerName}/", ""))
+                        .Where(w => _versionRegex.Replace(Path.GetFileName(w), "") == "")
+                        .ToList()
+                        .ForEach(f => TryDeleteBlob(f));
+
+                    // Drop the version file signaling such for any race-condition builds.
+                    CreateBlobIfNotExists(targetVersionFile);
+                }
+
+                CopyBlobs(sourceDir, channelDir);
+
+                // Generate the latest version text file
+                string versionText =
+                    $"{Commit}{Environment.NewLine}" +
+                    $"{ProductVersion}{Environment.NewLine}";
+
+                if (LatestVersionFilenames?.Any() != true)
+                {
+                    LatestVersionFilenames = DefaultLatestVersionFilenames;
+                }
+
+                foreach (string latestFilename in LatestVersionFilenames)
+                {
+                    PublishStringToBlob(
+                        ContainerName,
+                        $"{channelDir}{latestFilename}",
+                        versionText,
+                        "text/plain");
+                }
+            }
+            finally
+            {
+                Log.LogMessage($"Releasing lease on semaphore blob '{semaphoreBlob}'");
+                blobLease.Release();
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        public bool CopyBlobs(string sourceFolder, string destinationFolder)
```

Comment:
(minor) Do these methods need to be public?

Reply:
Forgot to look at these after the copy--they don't. Fixing.